### PR TITLE
Display developer measures

### PR DESCRIPTION
**Pull request by teryk**

### CODE REVIEW

**Comment by julienlancelot** [comment]
First small feedback : the name of the branch doesn't contain your initials :)

File: sonar-db/src/main/java/org/sonar/db/measure/MeasureDao.java [hunk 1]
```diff
@@ -109,11 +110,15 @@ public MeasureDto selectByComponentKeyAndMetricKey(DbSession session, String com
    * Used by plugin Developer Cockpit
    */
   public List<MeasureDto> selectBySnapshotIdsAndMetricIds(final DbSession dbSession, List<Long> snapshotIds, final List<Integer> metricIds) {
+    return selectByDeveloperAndSnapshotIdsAndMetricIds(dbSession, null, snapshotIds, metricIds);
+  }
+
+  public List<MeasureDto> selectByDeveloperAndSnapshotIdsAndMetricIds(final DbSession dbSession, @Nullable final Long developerId, List<Long> snapshotIds, final List<Integer> metricIds) {
```
Comment (SonarTech) [inline sonar-db/src/main/java/org/sonar/db/measure/MeasureDao.java]: MINOR Split this 187 characters long line (which is greater than 180 authorized). [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS00103)

File: sonar-db/src/main/java/org/sonar/db/measure/MeasureDao.java [hunk 2]
```diff
@@ -82,7 +83,7 @@ public MeasureDto selectByComponentKeyAndMetricKey(DbSession session, String com
   /**
    * Used by plugin Developer Cockpit
    */
-  public List<MeasureDto> selectByDeveloperForSnapshotAndMetrics(final DbSession dbSession, final long developerId, final long snapshotId, Collection<Integer> metricIds) {
+  public List<MeasureDto> selectByDeveloperForSnapshotAndMetrics(final DbSession dbSession, @Nullable final Long developerId, final long snapshotId, Collection<Integer> metricIds) {
```
Comment (SonarTech) [inline sonar-db/src/main/java/org/sonar/db/measure/MeasureDao.java]: MINOR Split this 181 characters long line (which is greater than 180 authorized). [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS00103)

**Comment by SonarTech** [comment]
#### Analysis summary
SonarQube analysis reported 3 issues
* MAJOR 1 major
* MINOR 2 minor

Watch the comments in this conversation to review them.

#### 1 unreported issues

Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:

1. org.sonarsource.sonarqube:sonar-process:target/generated-sources/protobuf/org/sonar/process/systeminfo/protobuf: MAJOR Add a 'package-info.java' file to document the 'protobuf' package [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1228)

File: server/sonar-server/src/main/java/org/sonar/server/measure/ws/ComponentTreeDataLoader.java [hunk 3]
```diff
@@ -101,6 +105,9 @@ ComponentTreeData load(ComponentTreeWsRequest wsRequest) {
           .setBaseComponent(baseComponent)
           .build();
       }
+      Long developerId = (wsRequest.getDeveloperId() == null && wsRequest.getDeveloperKey() == null)
```
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/measure/ws/ComponentTreeDataLoader.java]: Hard to read, I would move this to a dedicated method

File: server/sonar-server/src/main/java/org/sonar/server/measure/ws/ComponentTreeDataLoader.java [hunk 4]
```diff
@@ -148,7 +155,11 @@ ComponentTreeData load(ComponentTreeWsRequest wsRequest) {
...
 
   private ComponentDtosAndTotal searchComponents(DbSession dbSession, ComponentTreeQuery dbQuery, ComponentTreeWsRequest wsRequest) {
-    switch (wsRequest.getStrategy()) {
+    if (dbQuery.getQualifiers() != null && dbQuery.getQualifiers().isEmpty()) {
+      return new ComponentDtosAndTotal(Collections.<ComponentDtoWithSnapshotId>emptyList(), 0);
+    }
+    String strategy = requireNonNull(wsRequest.getStrategy());
+    switch (strategy) {
       case CHILDREN_STRATEGY:
         return new ComponentDtosAndTotal(
```
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/measure/ws/ComponentTreeDataLoader.java]: Not related to this PR, but this branch is not covered by UT...

File: server/sonar-server/src/test/java/org/sonar/server/measure/ws/ComponentTreeActionTest.java [hunk 5]
```diff
@@ -318,6 +321,107 @@ public void load_developer_descendants() {
...
+      .setParam(PARAM_STRATEGY, LEAVES_STRATEGY)
+      .setParam(PARAM_METRIC_KEYS, "ncloc")
+    );
+
+    assertThat(result.getBaseComponent().getId()).isEqualTo(projectUuid);
+    assertThat(result.getComponentsCount()).isEqualTo(0);
+  }
+
+  @Test
+  public void load_developer_measures_by_developer_key() {
```
Comment (julienlancelot) [inline server/sonar-server/src/test/java/org/sonar/server/measure/ws/ComponentTreeActionTest.java]: Missing same UT with unknown developer id

File: server/sonar-server/src/main/java/org/sonar/server/measure/ws/ComponentAction.java [hunk 6]
```diff
@@ -127,12 +131,15 @@ private ComponentWsResponse doHandle(ComponentWsRequest request) {
     DbSession dbSession = dbClient.openSession(false);
     try {
       ComponentDto component = componentFinder.getByUuidOrKey(dbSession, request.getComponentId(), request.getComponentKey(), COMPONENT_ID_AND_KEY);
+      Long developerId = request.getDeveloperId() == null && request.getDeveloperKey() == null
```
Comment (julienlancelot) [inline server/sonar-server/src/main/java/org/sonar/server/measure/ws/ComponentAction.java]: Should be moved to a method

**Comment by julienlancelot** [comment]
Some feedbacks to apply, then LGTM

**Comment by teryk** [comment]
the branch was used by Stas and myself, that's why there were not our initials : )

File: server/sonar-server/src/main/java/org/sonar/server/measure/ws/ComponentTreeDataLoader.java (hunk 3 above)
Comment (teryk) [inline server/sonar-server/src/main/java/org/sonar/server/measure/ws/ComponentTreeDataLoader.java]: ok

File: server/sonar-server/src/main/java/org/sonar/server/measure/ws/ComponentTreeDataLoader.java (hunk 4 above)
Comment (teryk) [inline server/sonar-server/src/main/java/org/sonar/server/measure/ws/ComponentTreeDataLoader.java]: very good point, thank you

File: server/sonar-server/src/test/java/org/sonar/server/measure/ws/ComponentTreeActionTest.java (hunk 5 above)
Comment (teryk) [inline server/sonar-server/src/test/java/org/sonar/server/measure/ws/ComponentTreeActionTest.java]: look line 325, you'll be surprised : )

File: server/sonar-server/src/main/java/org/sonar/server/measure/ws/ComponentAction.java (hunk 6 above)
Comment (teryk) [inline server/sonar-server/src/main/java/org/sonar/server/measure/ws/ComponentAction.java]: ok

File: server/sonar-server/src/test/java/org/sonar/server/measure/ws/ComponentTreeActionTest.java (hunk 5 above)
Comment (julienlancelot) [inline server/sonar-server/src/test/java/org/sonar/server/measure/ws/ComponentTreeActionTest.java]: I don't get your point, line 325 is load_developer_measures_by_developer_uuid, it's not an UT that check unknown developer id ???

**Pull request merged**